typescript/sdk/src/gas/calculator.ts

```diff
@@ -239,27 +254,66 @@ export class InterchainGasCalculator {
       ]),
     });
 
-    // directHandleCallGas includes the intrinsic gas
+    // Subtract intrinsic gas, which is included in directHandleCallGas.
+    // Note the "real" intrinsic gas will always be higher than this.intrinsicGas
+    // due to calldata costs, but this is desired because it results in a generous
+    // final estimate.
     return directHandleCallGas
-      .add(this.inboxProvingAndProcessingGas)
-      .add(this.messageGasEstimateBuffer);
+      .add(this.messageGasEstimateBuffer)
+      .sub(this.intrinsicGas);
+  }
+
+  /**
+   * @param originDomain The domain of the origin chain.
+   * @param destinationDomain The domain of the destination chain.
+   * @returns An estimated gas amount a relayer will spend when submitting a signed
+   * checkpoint to the destination domain.
+   */
+  async checkpointRelayGas(
+    originDomain: number,
+    destinationDomain: number,
+  ): Promise<BigNumber> {
+    // The gas used if the quorum threshold of a signed checkpoint is zero.
+    // Includes intrinsic gas and all other gas that does not scale with the
+    // number of signatures. Note this does not consider differences in intrinsic gas for
+    // different chains.
+    // Derived by observing the amount of gas consumed for a quorum of 1 (~86800 gas),
+    // subtracting the scaling gas per signature, and rounding up for safety.
+    const baseGasAmount = 80_000;
+    // Really observed to be about 8350, but rounding up for safety.
+    const gasPerSignature = 9_000;
+
+    const validatorManager = this.core.mustGetInboxValidatorManager(
+      originDomain,
+      destinationDomain,
+    );
+    const threshold = await validatorManager.threshold();
+
+    return threshold.mul(gasPerSignature).add(baseGasAmount);
   }
 
   /**
    * @returns A generous estimation of the gas consumption of all prove and process
-   * operations in Inbox.sol, excluding:
-   * 1. Intrinsic gas.
-   * 2. Any gas consumed within a `handle` function when processing a message once called.
+   * operations within Inbox.sol, including intrinsic gas. Does not include any gas
+   * consumed within a message's recipient `handle` function.
    */
-  get inboxProvingAndProcessingGas() {
+  get inboxProcessOverheadGas(): BigNumber {
```

Comment:
This eventually needs to account for sovereign consensus logic. Of course hard to predict what APi is necessary here, but maybe at least make it async?